dpe/src/commands/derive_child.rs

```diff
@@ -142,13 +142,7 @@ impl CommandExecution for DeriveChildCmd {
             dpe.generate_new_handle(env)?
         };
 
-        if !self.retains_parent() {
-            dpe.contexts[parent_idx].state = ContextState::Retired;
-            dpe.contexts[parent_idx].handle = ContextHandle([0xff; ContextHandle::SIZE]);
-        } else if !dpe.contexts[parent_idx].handle.is_default() {
-            dpe.contexts[parent_idx].handle = dpe.generate_new_handle(env)?;
-        }
-
+        let unactivated_child_context = dpe.contexts[child_idx];
```

Comment:
I think it would be more clear what's happening here if we make a `Context::default`, populate it, and then at the end, do `dpe.contexts[child_idx] = tmp_context;`

This allows you to still use the `?` mechanism for returns.

Reply:
Done.